### PR TITLE
Corrects error logging

### PR DESCRIPTION
**Pull request by ciroque**
Error logging was failing, this corrects that

### CODE REVIEW

File: test/request_cache/plug_test.exs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  use ExUnit.Case, async: true
+
+  defmodule FailingCacheModule do
+    def get(_cache_key) do
+      {:error, %{reason: :timeout}}
+    end
+  end
+
+  setup do
+    Application.put_env(:request_cache_plug, :request_cache_module, RequestCache.ConCacheStore)
```
Comment (RTLS) [inline test/request_cache/plug_test.exs]: why do we put this in env? I think it is overwritten for all the tests by the other `put_env`

File: test/request_cache/plug_test.exs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    test "it handles errors from the cache implementation on GraphQL endpoints" do
+      conn = %Plug.Conn{request_path: "/graphql", query_string: "?query=query MyQuery{myQuery{}}", method: "GET"}
+      assert %Plug.Conn{} = RequestCache.Plug.call(conn, nil)
+    end
+
+    test "it handles errors from the cache implementation on REST endpoints" do
+      conn = %Plug.Conn{request_path: "/my/object/1", query_string: "?page=1", method: "GET"}
+      assert %Plug.Conn{} = RequestCache.Plug.call(conn, nil)
+    end
```
Comment (RTLS) [inline test/request_cache/plug_test.exs]: should we use `capture_log` to verify the logging? probably not necessary to be honest

**Review by RTLS — APPROVED** [review]

File: test/request_cache/plug_test.exs (hunk 1 above)
Comment (MikaAK) [reply test/request_cache/plug_test.exs]: Yeah it is, this is also the default so I'm not sure why setting it makes a difference

Comment (ciroque) [reply test/request_cache/plug_test.exs]: Normally I would do this with `mox`, but that doesn't seem to be very popular. The outer `setup` resets the value for all the other tests, the `setup` local to the `describe` sets the failing module for just those tests.

File: test/request_cache/plug_test.exs (hunk 2 above)
Comment (ciroque) [reply test/request_cache/plug_test.exs]: :+1:

File: test/request_cache/plug_test.exs (hunk 1 above)
Comment (ciroque) [reply test/request_cache/plug_test.exs]: Removed

File: test/request_cache/plug_test.exs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    def get(_cache_key) do
+      {:error, %{reason: :timeout}}
+    end
+  end
+
+  describe "call/2" do
+    @expected_log_content "reason: :timeout"
+
+    setup do
+      Application.put_env(:request_cache_plug, :request_cache_module, FailingCacheModule)
```
Comment (MikaAK) [inline test/request_cache/plug_test.exs]: We probably should never overwrite this here it changes every future test after it, instead there's a cache key that's an option to the plug

**Review by MikaAK — APPROVED** [review]

**Pull request merged**